### PR TITLE
FIP-31360 Copy Lexis Nexis code from Financeit

### PR DESCRIPTION
**Pull request by jianaj**
- Add gem dependencies
- Port over LN code with bare minimum requirements

https://financeit.atlassian.net/browse/FIP-31360

### CODE REVIEW

File: lib/business_id.rb [hunk 1]
```diff
@@ -0,0 +1,139 @@
+module BusinessID
```
Comment (andyw8) [inline lib/business_id.rb]: There's a convention that everything is named within the gem name, i.e. `LexisNexis::BusinessID`.

This avoids the risk of clashes if two gems happen to use the same class name.

File: lexis_nexis.gemspec [hunk 2]
```diff
@@ -1,23 +1,21 @@
...
 Gem::Specification.new do |spec|
   spec.name          = "lexis_nexis"
   spec.version       = LexisNexis::VERSION
   spec.authors       = ["Jiana"]
   spec.email         = ["jjavier@financeit.io"]
 
-  spec.summary       = %q{TODO: Write a short summary, because RubyGems requires one.}
-  spec.description   = %q{TODO: Write a longer description or delete this line.}
-  spec.homepage      = "TODO: Put your gem's website or public repo URL here."
+  spec.summary       = "Financeit Lexis Nexis Integration"
```
Comment (andyw8) [inline lexis_nexis.gemspec]: I think it would useful to aim for to be a a general Lexis Nexis client, with nothing specific to Financeit. That should help to encourage a well-defined architectural boundary.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+image: "ruby:2.4.4"
```
Comment (jianaj) [inline .gitlab-ci.yml]: Not sure if I need this file committed, this project is currently not running on our CI

File: lexis_nexis.gemspec [hunk 4]
```diff
@@ -1,23 +1,21 @@
...
+  spec.homepage      = "https://github.com/financeit/lexis_nexis"
   spec.license       = "MIT"
 
   # Prevent pushing this gem to RubyGems.org. To allow pushes either set the 'allowed_push_host'
   # to allow pushing to a single host or delete this section to allow pushing to any host.
   if spec.respond_to?(:metadata)
-    spec.metadata["allowed_push_host"] = "TODO: Set to 'http://mygemserver.com'"
+    spec.metadata["allowed_push_host"] = 'http://www.financeit.ca'
```
Comment (jianaj) [inline lexis_nexis.gemspec]: Not quite sure what to put here, but didn't want to make it public so I put something

File: .rspec [hunk 5]
```diff
@@ -1,3 +1,3 @@
 --format documentation
 --color
---require spec_helper
+--require support/spec_helper
```
Comment (andyw8) [inline .rspec]: Any reason for putting this within `support`?

File: .gitlab-ci.yml (hunk 3 above)
Comment (andyw8) [reply .gitlab-ci.yml]: Let's leave this out for just now.

File: .rspec (hunk 5 above)
Comment (jianaj) [reply .rspec]: Not really, is it traditionally in the root `spec` folder? Just moved it because there is another helper file in `support` already but not a big deal to move it back

File: .gitlab-ci.yml (hunk 3 above)
Comment (jianaj) [reply .gitlab-ci.yml]: Nevermind, it is running. Not sure how exactly but... it works. Someone must have configured a runner for this project? I dont know but I'll take it!

File: lexis_nexis.gemspec [hunk 6]
```diff
@@ -1,38 +1,43 @@
...
-  spec.add_development_dependency "rake", "~> 10.0"
-  spec.add_development_dependency "rspec", "~> 3.0"
+  spec.add_development_dependency 'bundler', '~> 1.16'
+  spec.add_development_dependency 'pry-byebug'
+  spec.add_development_dependency 'rake', '~> 10.0'
+  spec.add_development_dependency 'rb-readline'
+  spec.add_development_dependency 'rspec', '~> 3.0'
+  spec.add_development_dependency 'rubocop'
+  spec.add_development_dependency 'rubocop-rspec'
+  spec.add_development_dependency 'savon'
```
Comment (andyw8) [inline lexis_nexis.gemspec]: `savon` should be runtime dependency: 

https://guides.rubygems.org/specification-reference/#add_development_dependency

File: lexis_nexis.gemspec [hunk 7]
```diff
@@ -1,38 +1,43 @@
...
-  spec.require_paths = ["lib"]
+  spec.require_paths = ['lib']
 
-  spec.add_development_dependency "bundler", "~> 1.16"
-  spec.add_development_dependency "rake", "~> 10.0"
-  spec.add_development_dependency "rspec", "~> 3.0"
+  spec.add_development_dependency 'bundler', '~> 1.16'
+  spec.add_development_dependency 'pry-byebug'
+  spec.add_development_dependency 'rake', '~> 10.0'
+  spec.add_development_dependency 'rb-readline'
```
Comment (andyw8) [inline lexis_nexis.gemspec]: Why is this needed?

File: spec/support/spec_helper.rb [hunk 8]
```diff
@@ -1,9 +1,15 @@
-require "bundler/setup"
-require "lexis_nexis"
+# frozen_string_literal: true
+
+require 'pry'
+require 'savon'
```
Comment (andyw8) [inline spec/support/spec_helper.rb]: We'll need to require savon from `lib/lexis_nexis`.

File: lexis_nexis.gemspec (hunk 7 above)
Comment (jianaj) [reply lexis_nexis.gemspec]: I was getting an error before that said I needed to have this with `pry` but I'm not getting it anymore and `pry` seems to be working so I'll remove

File: lexis_nexis.gemspec (hunk 4 above)
Comment (andyw8) [reply lexis_nexis.gemspec]: Since we're probably not going to publishing it on rubygems just yet, we can remove this section (i.e. from `if` to `end`).

File: .rspec (hunk 5 above)
Comment (andyw8) [reply .rspec]: Yeah, doesn't matter too much but `spec/` is just the convention.

File: bin/console [hunk 9]
```diff
@@ -1,7 +1,8 @@
 #!/usr/bin/env ruby
 
-require "bundler/setup"
-require "lexis_nexis"
+require 'bundler/setup'
+require 'lexis_nexis'
+require 'lexis_nexis/business_id'
```
Comment (andyw8) [inline bin/console]: There shouldn't a need to modify this. Think of `lib/lexis_nexis.rb` at the top of a tree of dependencies. From there you can require things you depend on, and they in turn can require things they depend on, etc.

File: lib/lexis_nexis/business_id.rb [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+    #options defaults
+    INCLUDE_MS_OVERRIDE = 0
+    INCLUDE_DL_VERIFICATION = 0
+    PO_BOX_COMPLIANCE = 0
+    GLOBAL_WATCHLIST_THRESHOLD = 0.84
+    BUSINESS_DEFENDER = 0
+    INCLUDE_ALL_RISK_INDICATORS = 0
+
+    def find_by_company_name(name, opts = {})
+      binding.pry
```
Comment (andyw8) [inline lib/lexis_nexis/business_id.rb]: accidentally left in?

Comment (jianaj) [reply lib/lexis_nexis/business_id.rb]: whoops!
